src/main/java/dev/streamx/sling/connector/ResourceInfo.java

```
@@ -32,8 +33,15 @@ public class ResourceInfo {
   public ResourceInfo(
       @JsonProperty("path") String path,
       @JsonProperty("primaryNodeType") String primaryNodeType) {
-    this.path = path;
-    this.primaryNodeType = primaryNodeType;
+    this.path = requireNotBlank("path", path);
+    this.primaryNodeType = requireNotBlank("primaryNodeType", primaryNodeType);
```

I think it's time to make this a requirement.
Until now, blank resource paths were silently skipped.
 - remove this, since blank path means something went wrong - leave earlier
 - no need to handle blank paths anymore in the main code, which got more complex - so remove this check from there